src/text.rs

```diff
@@ -63,8 +40,9 @@ impl<T: Read> super::Input for Input<T> {
         let send_data = sel.send(data_channel);
         let recv_ack = sel.recv(&self.ack_channel);
         let mut line_no = 0;
-        let mut line = Vec::new();
+
         'poll: loop {
+            let mut line = Vec::new();
```

Comment:
This causes a new `Vec` to be created in every iteration. Since `line`becomes an empty `Vec` in Line 88, we can just re-use `line` without creating a new `line`.